### PR TITLE
Implemented calculateClicks and calculateImpressions in src/utils

### LINKED ISSUE #2: Rewrite calculateClicksAndImpressions as a pure function
Function calculateClicksAndImpressions is doing too many things and has side effects. `const selectedSuggestion` and `const rawData` should be passed as arguments. Also, the function implementation could be replaced into the App.js component.

- This code should be generic

```
this.state = {
      suggestions: [],
      clicks: 0,
      impressions: 0
    };
```
```
this.setState({
      clicks: totalClicks,
      impressions: totalImpressions
    });
```

### PR DESCRIPTION
**Pull request by zeljko-djuric**
Fixes #2

### CODE REVIEW

File: src/utils/calculateClicks.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import _ from "lodash";
+
+const calculateClicks = (campaignOrChannel, data) => {
```
Comment (vlabs) [inline src/utils/calculateClicks.js]: DRY with calculateClicks and calculateImpressions

merge those two into one

File: src/utils/calculateClicks.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  const receivedData = data;
```
Comment (vlabs) [inline src/utils/calculateClicks.js]: this assignment is not necessary, directly use `data` with `filter`

File: src/utils/calculateClicks.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  const clicks = _.filter(receivedData, element => {
```
Comment (vlabs) [inline src/utils/calculateClicks.js]: use es6 `filter` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
chain functions: directly `sumBy` the result of `filter`

File: src/utils/calculateClicks.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    if (
+      _.isEqual(element.campaign, campaignOrChannel) ||
+      _.isEqual(element.channel, campaignOrChannel)
+    ) {
+      return element;
```
Comment (vlabs) [inline src/utils/calculateClicks.js]: both es6 and lodash `filter` expect a boolean. hence 
```
data.filter(item =>
   _.isEqual(item.campaign, campaignOrChannel) || _.isEqual(item.channel, campaignOrChannel)
)
``` 
is enough

File: src/utils/calculateClicks.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    }
+  });
+  const totalClicks = _.sumBy(clicks, "clicks");
+  return totalClicks;
```
Comment (vlabs) [inline src/utils/calculateClicks.js]: if u follow my request then you can avoid a `return` statement for this entire function

**Review by vlabs — CHANGES_REQUESTED** [review]

File: src/utils/calculateClicks.js (hunk 1 above)
Comment (zeljko-djuric) [reply src/utils/calculateClicks.js]: a01fb507ddd48c3aaf56b9f518982ad20b70dde2

File: src/utils/calculateClicks.js (hunk 2 above)
Comment (zeljko-djuric) [reply src/utils/calculateClicks.js]: a95a142a96ba9ec99ee2fc325f65cccf17c148cc

File: src/utils/calculateClicks.js (hunk 3 above)
Comment (zeljko-djuric) [reply src/utils/calculateClicks.js]: df73dcba84444ea4376252d33043dcbca7551c19

File: src/utils/calculateClicks.js (hunk 5 above)
Comment (zeljko-djuric) [reply src/utils/calculateClicks.js]: 0ad76fe2fd3d0221c48e0a55eda118f4fb6947a2

File: src/utils/calculateClicksAndImpressions.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import _ from "lodash";
+
+const calculateClicksAndImpressions = (campaignOrChannel, data, metricName) =>
```
Comment (vlabs) [inline src/utils/calculateClicksAndImpressions.js:3]: you can remove now all `calculate*` functions (and tests) except this one

but add tests for this one!

**Review by vlabs — CHANGES_REQUESTED** [review]

File: src/utils/calculateClicksAndImpressions.js (hunk 6 above)
Comment (zeljko-djuric) [reply src/utils/calculateClicksAndImpressions.js:3]: caf0a6411021cf487a19fa3e55383711464635b3

File: src/utils/testData.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    channel: "Display",
+    clicks: 312,
+    impressions: 642
+  }
+];
+
+const option1 = "DE|SN|Longboard|Generisch";
+const option2 = "Search";
+const option3 = "AT|SN|Snowboard|Generisch";
+// const option4 = "DE | SP Gadgets";
```
Comment (vlabs) [inline src/utils/testData.js]: either use option 4 to 15 or remove comments

File: src/utils/calculateClicks.js (hunk 4 above)
Comment (vlabs) [reply src/utils/calculateClicks.js]: done/outdated

File: src/utils/testData.js (hunk 7 above)
Comment (zeljko-djuric) [reply src/utils/testData.js]: 9f87f40e9771eba9a654b1f4fc1f595280510a3f

**Review by vlabs — APPROVED** [review]

**Pull request merged**